packages/control-group/src/index.tsx

```diff
@@ -0,0 +1,258 @@
+          activeGroup={Boolean(
+            Array.isArray(groupValue) ? groupValue.length : groupValue
+          )}
+          isOpen={isOpen}
+        >
+          {item.label}
+        </ControlButton>
+      )
+  }
+}
+
+const ControlGroupFC: React.FC<ControlGroupProps> = props => {
+  const {height, controlAppearance, groupAppearance} = useDefaults(
+    'controlGroup',
+    props,
+    {
+      height: 32,
+      controlAppearance: 'default' as ControlAppearanceType,
+      groupAppearance: 'button' as ControlGroupAppearanceType
+    }
+  )
+  const {
+    items,
+    onChange,
+    value,
+    layout,
+    textAlign,
+    disabled,
+    invalid,
+    visibleCount
+  } = props
+
+  const [controlsNumber, setControlsNumber] = React.useState(0)
+  const [isOpen, setIsOpen] = React.useState(false)
+  const [width, setWidth] = React.useState(0)
+  const hasMoreButton =
+    controlAppearance === 'outline' && groupAppearance === 'button'
+  const wrapperNode = React.useRef<HTMLDivElement>(null)
+  const moreButtonNode = React.useRef<HTMLButtonElement>(null)
+
+  React.useEffect(() => {
+    setControlsNumber(visibleCount ? visibleCount + 1 : 0)
+  }, [])
+
+  // TODO: add recalculate on window resize
+  React.useLayoutEffect(() => {
+    if (
+      wrapperNode &&
+      wrapperNode.current &&
+      !visibleCount &&
+      !controlsNumber
+    ) {
+      const {clientWidth, children: listChildren} = wrapperNode.current
+      let widthSum = moreButtonNode.current
+        ? moreButtonNode.current.clientWidth
+        : 0
+      let childCount = 0
+      Array.from(listChildren).forEach(child => {
+        if (widthSum + child.scrollWidth <= clientWidth) {
+          widthSum += child.scrollWidth
+          childCount++
+        }
+      })
+      setControlsNumber(childCount)
+    }
+    if (wrapperNode && wrapperNode.current) {
+      setWidth(wrapperNode.current.clientWidth)
+    }
+  }, [wrapperNode && wrapperNode.current && wrapperNode.current.clientWidth])
+
+  return (
+    <>
+      <ControlGroupWrapper
+        ref={wrapperNode}
+        groupAppearance={groupAppearance}
+        controlAppearance={controlAppearance}
+        childrenAmount={items.length}
+        layout={layout}
+        isOpen={isOpen}
+        visibleItemsCount={controlsNumber}
+        hasMoreButton={hasMoreButton}
+        width={width > 0 ? width : null}
+      >
+        {items.map(item => (
+          <Control
+            key={`${item.label}-${item.value}`}
+            item={item}
+            onChange={onChange}
+            value={value}
+            textAlign={textAlign}
+            disabled={disabled}
+            invalid={invalid}
+            groupAppearance={groupAppearance}
+            controlAppearance={controlAppearance}
+            height={height}
+            isOpen={isOpen}
+          />
+        ))}
+        {hasMoreButton && (
+          <>
+            <IconDecorator>
+              <Select
```

Comment:
There is some error with `full` (screenshot from the storybook):
<img width="1096" alt="Zrzut ekranu 2020-02-3 o 13 29 40" src="https://user-images.githubusercontent.com/8265332/73653355-4af11980-4689-11ea-93ec-0bd9866a32da.png">

Reply:
done